firestore/integration_test.go

```diff
@@ -1489,6 +1489,73 @@ func TestIntegration_CollectionGroupQueries(t *testing.T) {
 	}
 }
 
+func TestPartionQuery(t *testing.T) {
+	coll := collectionIDs.New()
+	ctx := context.Background()
+	h := testHelper{t}
+	client := integrationClient(t)
+	cr := client.Collection(coll)
+	subColl := coll + "-sub"
+	// Minimum partition size is 128.
+	documentCount := 128*2 + 127
+	var want []string
+	parents := []string{"abc-123", "def-456", "ghi-789"}
+	for i, parent := range parents {
+		dr1 := cr.Doc(parent)
+		scr := dr1.Collection(subColl)
+		for j := 0; j <= documentCount/3; j++ {
+			if i != len(parents)-1 || j != documentCount/3 {
```

Comment:
I believe these loops can be updated but I'm not sure how exactly

Reply:
Doesn't look like you actually need this if-statement. You're doing `for` intil `documentCount/3` - everything should be alright.

Reply:
It confuses me that we have such an odd number of documents `128*2 + 127`, and in the loop we get` 128*3`. I tried to do here as on python was https://github.com/googleapis/python-firestore/pull/209/files#diff-2a00933308324820f40f9649aa9fe3008280233f557d6b5cc4b01c48e79c65a7R932

Reply:
That probably caused by starting at 0 index. We need `num-1` to iterate `num` times

Reply:
Then we will have `127*3` documents not the same as `128*2 + 127` and minimum partition size is 128.